### PR TITLE
[enhancement]Add an option to ignore projected qty toward material requests

### PR DESCRIPTION
**Pull request by chaoyee**
If checked, Projected qty will be ignored. If material requests are generated based on sales orders, the qty reserved by sales orders will be included in projected qty, which causes the qty deduction of the material requests. Thus, ignorance of projected qty should be checked in this situation.

### CODE REVIEW

File: erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.json [hunk 1]
```diff
@@ -646,6 +646,33 @@
...
+   "depends_on": "create_material_requests", 
+   "description": "If checked, Projected qty will be ignored. If material requests are generated based on sales orders, the qty reserved by sales orders will be included in projected qty, which causes the qty deduction of the material requests. Thus, ignorance of projected qty should be checked in this situation.", 
+   "fieldname": "ignore_projected_qty", 
+   "fieldtype": "Check", 
+   "hidden": 0, 
+   "ignore_user_permissions": 0, 
+   "ignore_xss_filter": 0, 
+   "in_filter": 0, 
+   "in_list_view": 0, 
+   "label": "Ignore Projected Qty",
```
Comment (nabinhait) [inline erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.json]: "Ignore Projected Qty" label does not explain the usage of it. How about "Create Material Requests for All Required Qty"?

Comment (chaoyee) [inline erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.json]: Sounds great!
Let's do it!

Comment (nabinhait) [inline erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.json]: Please make the changes and resent the PR

Comment (chaoyee) [inline erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.json]: "ignore_projected_qty" has been changed to "create_material_requests_for_all_required_qty".

File: erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py [hunk 2]
```diff
@@ -394,7 +394,12 @@ def get_requested_items(self):
 			total_qty = sum([flt(d[0]) for d in so_item_qty])
 			if total_qty > item_projected_qty.get(item, 0):
 				# shortage
-				requested_qty = total_qty - flt(item_projected_qty.get(item))
+				if self.create_material_requests_for_all_required_qty:
```
Comment (anandpdoshi) [inline erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py]: Should this depend on the condition in line 395: `if total_qty > item_projected_qty.get(item, 0):`?

**Comment by nabinhait** [comment]
Added the feature via https://github.com/frappe/erpnext/pull/5327

File: erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py (hunk 2 above)
Comment (chaoyee) [inline erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py]: Either ways, both can reach the same result.

The "minimum order qty" is processed at:
[https://github.com/frappe/erpnext/blob/develop/erpnext/manufacturing/doctype/production_planning_tool/production_planning_tool.py#L400-L401]

**Comment by chaoyee** [comment]
Thanks!

**Pull request closed without merging**